### ISSUE TITLE
Clippy warnings

### ISSUE THREAD

**Issue by mlund** [simmons/cbm#4]
`cargo clippy` emits a series of warnings, most not too serious. This one sticks out, though:
~~~
error: using `self.to_string` in `fmt::Display` implementation will cause infinite recursion
  --> src/disk/validation.rs:60:31
   |
60 |             _ => f.write_str(&self.to_string()),
   |                               ^^^^^^^^^^^^^^^^
   |
   = note: `#[deny(clippy::recursive_format_impl)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#recursive_format_impl
~~~
Perhaps enable clippy as a Github action to automatically check commits? https://github.com/marketplace/actions/rust-clippy-check

**Comment by simmons** [comment]
Yes, there was another infinite recursion possibility that was introduced by the migration to the 2021 edition. I've opened PR #5 to address Clippy issues. When that's merged, we can look into enabling a GitHub action.

**Comment by mlund** [comment]
Closed by #5
